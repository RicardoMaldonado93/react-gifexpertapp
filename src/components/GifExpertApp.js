import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] })=>{
    
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategory={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category =>
                        <GifGrid 
                            category={category}
                            key={category}
                        />
                    )
                }

            </ol>
            
        </>
    );
}

export default GifExpertApp;