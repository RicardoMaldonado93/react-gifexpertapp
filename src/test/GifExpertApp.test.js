import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from "../components/GifExpertApp"

describe("Unit test for 'GiftExpertApp'", () => {


    test("should render correctly component", () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test("should view list of categories", () => {
        const categories = ["One Punch","Dragon Ball"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    });
    

})
