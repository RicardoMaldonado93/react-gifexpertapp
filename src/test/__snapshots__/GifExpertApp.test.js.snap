// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for 'GiftExpertApp' should render correctly component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Unit test for 'GiftExpertApp' should view list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
